storage/innobase/trx/trx0trx.cc

```diff
@@ -64,9 +64,15 @@ int thd_deadlock_victim_preference(const MYSQL_THD thd1, const MYSQL_THD thd2);
 
 /** The bit pattern corresponding to TRX_ID_MAX */
 const char trx_id_max_bytes[8] = {
-	'\377', '\377', '\377', '\377', '\377', '\377', '\377', '\377'
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff
 };
 
+/** The bit pattern corresponding to max timestamp */
+const char timestamp_max_bytes[7] = {
+	0x7f, 0xff, 0xff, 0xff, 0x0f, 0x42, 0x3f
```

Comment:
Nothing to fix right now, but this could not work on big-endian architectures. Buildbots will show.

Reply:
This is storage format, not machine format.